src/components/SearchReps.vue

```diff
@@ -195,23 +182,47 @@ export default {
             this.step2 = step2
         },
         CheckInputContent: function () {
-          if (this.postalCode !== '') {
-              this.hasContent = true
-          } else {
-              this.hasContent = false
-          }
+
+            if (this.searchText !== '') {
+                this.hasContent = true
+            } else {
+                this.hasContent = false
+            }
         },
         async CreateRepList() {
             try {
-                this.$store.commit('setGenericValue', { key: 'zipcode', value: this.postalCode })
-                const res = await axios.get(
-                    '/api/representatives/' + this.postalCode
+                this.$store.commit('setGenericValue', { key: 'searchText', value: this.searchText })
+                // check postal code is valid with regex
+                let res = ''
+                let params = {streetAddress: this.searchText}
+                let isPostalCodeValid =  /(^\d{5}$)|(^\d{5}-\d{4}$)/.test(this.searchText);
+                let streetAddressValid = /^(\d{3,})\s?(\w{0,5})\s([a-zA-Z]{2,30})\s([a-zA-Z]{2,15})\.?\s?(\w{0,5})$/.test(this.searchText);
+
+
+                if(isPostalCodeValid) {
+                 // console.log('valid input to get representatives with postal code')
+                  res = await axios.get(
+                    '/api/representatives/' + this.searchText
+                )
+                }
+
+                else if(streetAddressValid) {
+                  // console.log('valid input to get representatives with street address')
+                  res = await axios.get(
+                    '/api/representatives/' + ' ', {
+                        params
+                    }
```

Comment:
Just formatting - no need to add constants together instead of just combining them

```suggestion
                    '/api/representatives/ ', { params }
```